### ISSUE TITLE
feature requrest:  ignore option

### ISSUE THREAD

**Issue by mleathem** [validator/validator#213]
Would be great if there was an option to skip validation for part of the document.
There are times when developers know html is invalid, but for whatever reason cannot change this. 
use case:  
1) code out of developer's control (3rd party etc).
2) temporary development code, that can be ignored.

Implementation suggestion...   data-validate="false"

e.g. 
`<section data-validate="false">`
 `<!-- some known invalid code here would be skipped from validation  -->`
`</section>`

**Comment by mleathem** [comment]
PS. keep up the good work!

**Comment by zcorpan** [comment]
I don't think this should be controllable from the markup alone, since that will lead people to put the attribute on `<html>` to satisfy their client/boss/self.

https://validator.nu/ actually has this feature for XML documents, the "XMLNS filter". Maybe that can be tweaked to be a selector?

**Comment by sideshowbarker** [comment]
> https://validator.nu/ actually has this feature for XML documents, the "XMLNS filter". Maybe that can be tweaked to be a selector?

Yes, see https://github.com/validator/validator/blob/master/src/nu/validator/xml/NamespaceDroppingXMLReaderWrapper.java#L127

So I think it would be fairly simple to implement a similar content-dropping filter that instead based the dropping on a particular attribute name/value (e.g. `data-validate=false` or even just `data-novalidate` with no value).

**Comment by kurtseebauer** [comment]
Is this feature also available for HTML-5 documents? I'm using thymeleaf templates which are basically html conform but with additional th:*-attributes.
The xml namespace is declared in the html (<html lang="de" xmlns:th="http://www.thymeleaf.org">) but I can't find a way to make the validator ignore the attributes.

**Comment by zcorpan** [comment]
HTML doesn't support namespace syntax so that doesn't work. This issue is about adding some other mechanism to ignore subtrees in HTML.

That said, if you are using custom markup for templates, maybe you could use the `template` element  (which already allows anything inside, so long as there are no parse errors)?

**Comment by dijitali** [comment]
Would be great if any solution also supported targeted ignores for CSS and other file types beyond HTML.

For my use-case, I've ran into a problem where the CSS validator doesn't support a valid CSS feature (https://github.com/w3c/css-validator/issues/431).

**Comment by sideshowbarker** [comment]
This is not a feature that I ever plan to work on implementing myself, and nobody else has ever stepped forward to work on it. So I’m going ahead and closing this.
